### PR TITLE
95 create graphics system

### PR DESCRIPTION
**Pull request by osvegn**
New feature:
- Create SFMLSystem library.
- Create DrawComponent system to display Graphical Component.
- Create InputManagement system to enqueue function depending on SFML inputs.

Update feature:
- Remove unused <iostream> include in world.hpp.
- Add a "LAYER_NUMBER" enum in layer_e to know the max value of this enum.

Fix feature:
- Fix target_include_directory in library gameSystem.
- Fix shared resource RenderWindow by using the style passed as parameter.

### CODE REVIEW

File: libs/SFMLComponents/LayerLvL.hpp [hunk 1]
```diff
@@ -17,7 +17,7 @@ namespace ecs
     class LayerLvL : public Component {
       public:
         /// @brief Enumeration of the different layers can be used
-        enum layer_e { UNKNOWN, BACKGROUND, DECORATION, OBSTACLE, ENEMY, PLAYER, PROJECTILE };
+        enum layer_e { UNKNOWN, BACKGROUND, DECORATION, OBSTACLE, ENEMY, PLAYER, PROJECTILE, LAYER_NUMBER};
```
Comment (31Nathan) [inline libs/SFMLComponents/LayerLvL.hpp:20]: Can it be just Layer Level ?

File: libs/SFMLResource/CMakeLists.txt [hunk 2]
```diff
@@ -11,7 +11,7 @@ set(SRC
 add_library(${TARGET} SHARED ${SRC})
 
 target_include_directories(${TARGET}
-    PUBLIC ${SFML_INPUT_INC_ROOT}
+    PUBLIC ${INC_SFML_RESOURCE_ROOT}
```
Comment (31Nathan) [inline libs/SFMLResource/CMakeLists.txt:14]: Why ?

File: libs/SFMLSystems/DrawComponents.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#include "GraphicsTextComponent.hpp"
+#include "GraphicsRectangleComponent.hpp"
+
+using namespace ecs;
+
+void DrawComponents::run(World &world)
+{
+    std::vector<std::shared_ptr<Entity>> Inputs = world.joinEntities<LayerLvL>();
+
+    world.getResource<RenderWindowResource>().window.clear();
```
Comment (31Nathan) [inline libs/SFMLSystems/DrawComponents.cpp:21]: You're a genius, nice one !

File: libs/SFMLSystems/InputManagement.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+#include "ActionQueueComponent.hpp"
+#include "AllowControllerComponent.hpp"
+#include "AllowMouseAndKeyboardComponent.hpp"
+
+using namespace ecs;
+
+void InputManagement::run(World &world)
+{
+    sf::Event event;
+    std::vector<std::shared_ptr<Entity>> Inputs = world.joinEntities<MouseInputComponent, KeyboardInputComponent, ControllerButtonInputComponent, ControllerJoystickInputComponent, ActionQueueComponent>();
```
Comment (31Nathan) [inline libs/SFMLSystems/InputManagement.cpp:24]: Beware ! Join Entities isn't an "This component or This one" it's like an AND "This component and This one"

**Review by 31Nathan — COMMENTED** [review]
Just need few answers, otherwise good job

File: libs/SFMLResource/CMakeLists.txt (hunk 2 above)
Comment (osvegn) [reply libs/SFMLResource/CMakeLists.txt:14]: It wasn't the good variable previously

File: libs/SFMLSystems/InputManagement.cpp (hunk 4 above)
Comment (osvegn) [reply libs/SFMLSystems/InputManagement.cpp:24]: I know it, the entity must have all these components.

File: libs/SFMLComponents/LayerLvL.hpp (hunk 1 above)
Comment (osvegn) [reply libs/SFMLComponents/LayerLvL.hpp:20]: All right here

**Review by 31Nathan — DISMISSED** [review]
Seems good to me, will you have any tests ?

**Review by LucasTesnier — APPROVED** [review]
NTR!

**Comment by osvegn** [comment]
> Seems good to me, will you have any tests ?

Difficult to test because I work with the SFML library ;)

File: libs/SFMLSystems/DrawComponents.cpp [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    for (std::size_t layerI = 1; layerI < LayerLvL::LAYER_NUMBER; layerI++) {
+        auto layer = [&layerI, &world](std::shared_ptr<Entity> entityPtr) {
+            if (entityPtr->getComponent<LayerLvL>().layer == layerI) {
+                if (entityPtr->contains<GraphicsRectangleComponent>())
+                    world.getResource<RenderWindowResource>().window.draw(
+                        entityPtr->getComponent<GraphicsRectangleComponent>().shape);
+                if (entityPtr->contains<GraphicsTextComponent>())
+                    world.getResource<RenderWindowResource>().window.draw(
+                        entityPtr->getComponent<GraphicsTextComponent>().text);
+            }
```
Comment (ZQUEMA) [inline libs/SFMLSystems/DrawComponents.cpp:31]: This should be optimize later 
It's linked to the issue #129

**Review by ZQUEMA — APPROVED** [review]

**Review by LucasTesnier — APPROVED** [review]

**Pull request merged**